src/fpsemi-examples.cpp

```diff
@@ -621,135 +634,145 @@ namespace libsemigroups {
-      }
-    }
-    // R2
-    result.emplace_back(x ^ 3, x);
-    // R3
-    result.emplace_back(x * s[1], x);
-    result.emplace_back(s[1] * x, x);
-    // R4
-    result.emplace_back(x * s[2] * x, x * s[2] * x * s[2]);
-    result.emplace_back(x * s[2] * x * s[2], s[2] * x * s[2] * x);
-    result.emplace_back(s[2] * x * s[2] * x, x * s[2] * (x ^ 2));
-    result.emplace_back(x * s[2] * (x ^ 2), (x ^ 2) * s[2] * x);
-    if (n == 3) {
-      return result;
-    }
-    // R5
-    word_type const sigma = s[2] * s[3] * s[1] * s[2];
-    result.emplace_back((x ^ 2) * sigma * (x ^ 2) * sigma,
-                        sigma * (x ^ 2) * sigma * (x ^ 2));
-    result.emplace_back(sigma * (x ^ 2) * sigma * (x ^ 2),
-                        x * s[2] * s[3] * s[2] * x);
-    // R6
-    std::vector<word_type> l = {{}, {}, x * s[2] * s[1]};
-    for (size_t i = 3; i <= n - 1; ++i) {
-      l.push_back(s[i] * l[i - 1] * s[i] * s[i - 1]);
-    }
-    std::vector<word_type> y = {{}, {}, {}, x};
-    for (size_t i = 4; i <= n; ++i) {
-      y.push_back(l[i - 1] * y[i - 1] * s[i - 1]);
-    }
-    for (size_t i = 3; i <= n - 1; ++i) {
-      result.emplace_back(y[i] * s[i] * y[i], s[i] * y[i] * s[i]);
-    }
-    if (n == 4) {
+      // R1
+      for (size_t i = 1; i <= n - 1; ++i) {
+        for (size_t j = 1; j <= n - 1; ++j) {
+          result.emplace_back((s[i] * s[j]) ^ mij(i, j), e);
+        }
+      }
+      // R2
+      result.emplace_back(x ^ 3, x);
+      // R3
+      result.emplace_back(x * s[1], x);
+      result.emplace_back(s[1] * x, x);
+      // R4
+      result.emplace_back(x * s[2] * x, x * s[2] * x * s[2]);
+      result.emplace_back(x * s[2] * x * s[2], s[2] * x * s[2] * x);
+      result.emplace_back(s[2] * x * s[2] * x, x * s[2] * (x ^ 2));
+      result.emplace_back(x * s[2] * (x ^ 2), (x ^ 2) * s[2] * x);
+      if (n == 3) {
+        return result;
+      }
+      // R5
+      word_type const sigma = s[2] * s[3] * s[1] * s[2];
+      result.emplace_back((x ^ 2) * sigma * (x ^ 2) * sigma,
+                          sigma * (x ^ 2) * sigma * (x ^ 2));
+      result.emplace_back(sigma * (x ^ 2) * sigma * (x ^ 2),
+                          x * s[2] * s[3] * s[2] * x);
+      // R6
+      std::vector<word_type> l = {{}, {}, x * s[2] * s[1]};
+      for (size_t i = 3; i <= n - 1; ++i) {
+        l.push_back(s[i] * l[i - 1] * s[i] * s[i - 1]);
+      }
+      std::vector<word_type> y = {{}, {}, {}, x};
+      for (size_t i = 4; i <= n; ++i) {
+        y.push_back(l[i - 1] * y[i - 1] * s[i - 1]);
+      }
+      for (size_t i = 3; i <= n - 1; ++i) {
+        result.emplace_back(y[i] * s[i] * y[i], s[i] * y[i] * s[i]);
+      }
+      if (n == 4) {
+        return result;
+      }
+      // R7
+      for (size_t i = 4; i <= n - 1; ++i) {
+        result.emplace_back(x * s[i], s[i] * x);
+      }
       return result;
+    } else {
+      LIBSEMIGROUPS_EXCEPTION("not yet implemented");
```

Comment:
Maybe it'd be most helpful to re-iterate what the possible valid argument combinations are here?